### ISSUE TITLE
Incorrect error messages for cv2.ellipse

### ISSUE THREAD

**Issue by Erotemic** [opencv/opencv#12950]
##### System information (version)

```python
In [91]: cv2.__version__
Out[91]: '3.4.1'
In [92]: cv2
Out[92]: <module 'cv2.cv2' from '/home/joncrall/venv3.6/lib/python3.6/site-packages/cv2/cv2.cpython-36m-x86_64-linux-gnu.so'>
```
- Operating System / Platform => Ubuntu 18.04

##### Detailed description

The error message given by cv2.ellipse is incorrect. When I give it the wrong input types it complains about the number of arguments: `TypeError: ellipse() takes at most 5 arguments (8 given)`.  Ellipse certainly takes more than 5 args if you look at the docs: 

```
ellipse(...)
    ellipse(img, center, axes, angle, startAngle, endAngle, color[, thickness[, lineType[, shift]]]) -> img
    .   @brief Draws a simple or thick elliptic arc or fills an ellipse sector.
```

##### Steps to reproduce

```python
        import numpy as np
        import cv2
        img = np.zeros((10, 10,), dtype=np.uint32)
        center = (5, 5)
        axes = (2, 2)
        color = (1, 1, 1)
        cv2.ellipse(img, center, axes, 0.0, 0.0, 360.0, color, -1)
```

results in 

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-94-0d726ae218aa> in <module>()
      5 axes = (2, 2)
      6 color = (1, 1, 1)
----> 7 cv2.ellipse(img, center, axes, 0.0, 0.0, 360.0, color, -1)

TypeError: ellipse() takes at most 5 arguments (8 given)
```

If you change the code by correcting uint32 to uint8, then it works

```python
        import numpy as np
        import cv2
        img = np.zeros((10, 10,), dtype=np.uint8)
        center = (5, 5)
        axes = (2, 2)
        color = (1, 1, 1)
        cv2.ellipse(img, center, axes, 0.0, 0.0, 360.0, color, -1)
```

```
array([[0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 1, 0, 0, 0, 0],
       [0, 0, 0, 0, 1, 1, 1, 0, 0, 0],
       [0, 0, 0, 1, 1, 1, 1, 1, 0, 0],
       [0, 0, 0, 0, 1, 1, 1, 0, 0, 0],
       [0, 0, 0, 0, 0, 1, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
```

Thus, the error message produced by cv2.ellipse should be checked and fixed to better report what is actually going on. The issue has noting to do with number of arguments. It is the type of the input array that is unexpected. (~~Additionally, I think it wouldn't be a bad idea to make ellipse work with the uint32 type, because it does works with int32~~).

**Comment by dan-masek** [comment]
This kind of misleading error message is common to all Python bindings that resolve between multiple overloads of a C++ function. IIRC it tries to match the parameters for each of them, and if all fail, it produces an error based on the last attempt.

**Comment by VadimLevin** [comment]
Resolved in #19312.
```python
import numpy as np
import cv2
img = np.zeros((10, 10,), dtype=np.uint32)
center = (5, 5)
axes = (2, 2)
color = (1, 1, 1)
cv2.ellipse(img, center, axes, 0.0, 0.0, 360.0, color, -1)
```
produces the following error:
```
cv2.error: OpenCV(3.4.15-dev) :-1: error: (-5:Bad argument) in function 'ellipse'
> Overload resolution failed:
>  - img data type = 6 is not supported
>  - Expected Ptr<cv::UMat> for argument 'img'
>  - ellipse() takes at most 5 arguments (8 given)
>  - ellipse() takes at most 5 arguments (8 given)
```
